realm/realm-library/src/main/java/io/realm/RealmCache.java

```diff
@@ -248,4 +261,49 @@ static synchronized void invokeWithGlobalRefCount(RealmConfiguration configurati
         }
         callback.onResult(totalRefCount);
     }
+
+    /**
+     * Copies Realm database file from Android asset directory to the directory given in the {@link RealmConfiguration}.
+     * Copy is performed only at the first time when there is no Realm database file.
+     *
+     * @param configuration configuration object for Realm instance.
+     * @throws IOException if copying the file fails.
+     */
+    private static void copyAssetFile(RealmConfiguration configuration) {
+        File realmFile = new File(configuration.getRealmFolder(), configuration.getRealmFileName());
+        if (!realmFile.exists()) {
+            InputStream inputStream = null;
+            FileOutputStream outputStream = null;
+            try {
+                inputStream = configuration.getAssetFile();
+                if (inputStream != null) {
+                    outputStream = new FileOutputStream(realmFile);
+                    byte[] buf = new byte[1024];
+                    int bytesRead;
+                    while ((bytesRead = inputStream.read(buf)) > -1) {
+                        outputStream.write(buf, 0, bytesRead);
+                    }
+                    outputStream.close();
+                    inputStream.close();
+                }
+            } catch (IOException e) {
+                throw new RealmIOException("Could not resolve the path to the Realm asset file.");
+            } finally {
+                if (inputStream != null) {
+                    try {
+                        inputStream.close();
+                    } catch (IOException e) {
+                        // Ignore this exception because any significant errors should already have been handled
```

Comment:
Yes, throwing if outputStream fail makes a lot of sense. As @beeender says, if the Realm file is not in a proper state, all behavior is undefined.